tests/numerical_test/execute.py

```diff
@@ -161,16 +143,60 @@ def save_np_data(fpath: str, data):
     return res
 
 
+class BRTBackend:
+    def __init__(self, device, brt_file_path):
+        from torch.cuda.memory import caching_allocator_alloc, caching_allocator_delete
+        _allocator_alloc = caching_allocator_alloc if device == "cuda" else None
+        _allocator_delete = caching_allocator_delete if device == "cuda" else None
+        _stream = torch.cuda.current_stream()._as_parameter_.value if device == "cuda" else None
+        self.session = brt.Session(device=device.upper(),
+                                   alloc_func=_allocator_alloc,
+                                   free_func=_allocator_delete)
+        self.session.load(brt_file_path)
+        self.req = self.session.new_request_context(_stream)
+
+    def execute(self, inputs, outputs):
+        for offset, arg in zip(self.session.get_input_arg_offsets(), inputs):
+            assert list(self.session.get_static_shape(offset)) == list(arg.shape)
+            self.req.bind_arg(offset, arg.data_ptr())
+        for offset, ret in zip(self.session.get_output_arg_offsets(), outputs):
+            assert list(self.session.get_static_shape(offset)) == list(ret.shape)
+            self.req.bind_arg(offset, ret.data_ptr())
+        self.req.finish_io_binding()
+        self.req.run()
+        self.req.sync()
+
+    def profile(self, inputs, outputs, warmup_trials=10, run_trials=50):
+        for offset, arg in zip(self.session.get_input_arg_offsets(), inputs):
+            assert list(self.session.get_static_shape(offset)) == list(arg.shape)
+            self.req.bind_arg(offset, arg.data_ptr())
+        for offset, ret in zip(self.session.get_output_arg_offsets(), outputs):
+            assert list(self.session.get_static_shape(offset)) == list(ret.shape)
+            self.req.bind_arg(offset, ret.data_ptr())
+        self.req.finish_io_binding()
+        
```

Comment:
It might be better to encapsulate the binding of inputs/outputs as a class member method.

Reply:
I think it's not necessary to do it in e2e_testing, I prefer to consider how to support dynamic shape and unify the testing code of static shape & dynamic shape.